src/sql/workbench/parts/dataExplorer/browser/connectionViewletPanel.ts

```diff
@@ -128,6 +128,7 @@ export class ConnectionViewletPanel extends ViewletPanel {
 
 	dispose(): void {
 		this._serverTreeView.dispose();
+		this.objectExplorerService.unregisterServerTreeView();
```

Comment:
>this.objectExplorerService.unregisterServerTreeView(); [](start = 2, length = 54)

We're calling dispose on the tree view twice now. I'm not sure which one I prefer to do the disposing though....TBH this is where some sort of ref counting thing would probably be best. But if I had to choose I'd say leave the panel here in charge of disposing since it created it in the first place.

Reply:
Honestly, we can also not dispose off the serverTreeView if we know it's a core part of the data explorer which will always exist (just show/hidden). I prefer not disposing the serverTreeView at all. If that makes sense, I'll make a new pull request since this one is too cluttered.